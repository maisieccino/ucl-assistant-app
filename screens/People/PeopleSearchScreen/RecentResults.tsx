import React from "react"
import { View } from "react-native"
import Button from "../../../components/Button"
import SearchResult from "../../../components/SearchResult"
import { CentredText, SubtitleText } from "../../../components/Typography"
import type { Person } from "../../../types/uclapi"

interface RecentResultProp {
  recent: Person,
  viewPerson: () => void,
}

const RecentResult: React.FC<RecentResultProp> = ({ recent, viewPerson }) => (recent ? (
  <SearchResult
    key={recent.email}
    topText={recent.name}
    bottomText={recent.department}
    type="person"
    onPress={viewPerson}
  />
) : null)

interface RecentResultsProps {
  viewPerson: (e: string) => () => void,
  recents: Person[],
  clearRecentResults: () => void,
}

const RecentResults: React.FC<RecentResultsProps> = ({
  recents = [],
  clearRecentResults,
  viewPerson,
}) => {
  if (recents.length === 0) {
    return null
  }
  return (
    <View>
      <SubtitleText>Recently Searched</SubtitleText>
      {recents.map((recent) => (
        <RecentResult
          key={`${recent.email}-${recent.name}`}
          recent={recent}
          viewPerson={viewPerson(recent.email)}
        />
      ))}
      {recents.length > 0 ? (
        <Button onPress={clearRecentResults}>Clear</Button>
      ) : (
        <CentredText>Recent results will appear here.</CentredText>
      )}
    </View>
  )
}

export default RecentResults
