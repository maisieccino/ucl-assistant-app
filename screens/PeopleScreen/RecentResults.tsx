import React from "react"
import { View } from "react-native"
import { connect } from "react-redux"
import { generate } from "shortid"

import {
  clearRecentResults as clearRecentsResultsAction,
} from "../../actions/peopleActions"
import Button from "../../components/Button"
import SearchResult from "../../components/SearchResult"
import { CentredText, SubtitleText } from "../../components/Typography"
import type { AppStateType } from '../../configureStore'

interface Props {
  clearRecentResults: () => void,
  navigation: any,
  recents: any,
}

export class RecentResults extends React.Component<Props> {
  static mapDispatchToProps = (dispatch): any => ({
    clearRecentResults: (): any => dispatch(clearRecentsResultsAction()),
  })

  static mapStateToProps = (state: AppStateType): any => ({
    recents: state.people.recents,
  })

  static defaultProps = {
    clearRecentResults: (): void => { },
    recents: [],
  }

  viewPerson = (person) => (): void => {
    const { navigation } = this.props
    navigation.navigate(`PersonDetail`, person)
  }

  renderRecent = (recent = null) => {
    if (recent === null) {
      return null
    }
    return (
      <SearchResult
        key={generate()}
        topText={recent.name}
        bottomText={recent.department}
        type="person"
        onPress={this.viewPerson(recent)}
      />
    )
  }

  render(): React.ReactElement {
    const { recents, clearRecentResults } = this.props
    if (recents.length === 0) {
      return null
    }
    return (
      <View>
        <SubtitleText>Recently Searched</SubtitleText>
        {recents.map(this.renderRecent)}
        {recents.length > 0 ? (
          <Button onPress={clearRecentResults}>Clear</Button>
        ) : (
            <CentredText>Recent results will appear here.</CentredText>
        )}
      </View>
    )
  }
}

export default connect(
  RecentResults.mapStateToProps,
  RecentResults.mapDispatchToProps,
)(RecentResults)
