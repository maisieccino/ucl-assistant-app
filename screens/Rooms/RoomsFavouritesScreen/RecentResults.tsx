import { StackNavigationProp } from '@react-navigation/stack'
import React, { useCallback } from "react"
import { StyleSheet, View } from "react-native"
import { connect, ConnectedProps } from "react-redux"
import Button from "../../../components/Button"
import SearchResult from "../../../components/SearchResult"
import { CentredText, SubtitleText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import { clearRecents } from "../../../redux/actions/roomsActions"
import type { Room } from '../../../types/uclapi'
import type { RoomsNavigatorParamList } from "../RoomsNavigator"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 40,
  },
})

interface Props extends PropsFromRedux {
  navigation: StackNavigationProp<RoomsNavigatorParamList>,
}

const RecentResults: React.FC<Props> = ({
  recents = [],
  clearRecentRooms,
  navigation,
}) => {
  const navigateToRoomDetail = useCallback(
    (room: Room) => () => navigation.navigate(`RoomsDetail`, { room }),
    [navigation],
  )

  if (recents.length === 0) {
    return null
  }
  return (
    <View style={styles.container}>
      <SubtitleText>Recently Searched</SubtitleText>
      {recents.map((res) => (
        <SearchResult
          key={`${res.siteid}---${res.roomid}`}
          topText={res.roomname}
          bottomText={res.classification_name}
          type="location"
          onPress={navigateToRoomDetail(res)}
        />
      ))}
      {recents.length > 0 ? (
        <Button onPress={clearRecentRooms}>Clear</Button>
      ) : (
        <CentredText>Recent results will appear here.</CentredText>
      )}
    </View>
  )
}

const connector = connect(
  (state: AppStateType) => ({
    recents: state.rooms.recents,
  }),
  (dispatch) => ({
    clearRecentRooms: () => dispatch(clearRecents()),
  }),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(RecentResults)
