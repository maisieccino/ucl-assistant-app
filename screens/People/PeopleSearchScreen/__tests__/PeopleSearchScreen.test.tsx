import { cleanup, fireEvent, render } from '@testing-library/react-native'
import React from "react"
import "react-native"
import RecentResults from "../RecentResults"

describe(`RecentResults`, () => {
  beforeEach(() => {
    jest.useRealTimers()
    jest.clearAllMocks()
  })

  afterEach(() => {
    cleanup()
  })

  it(`renders an empty RecentResults component`, async () => {
    const props = {
      clearRecentResults: jest.fn(),
      recents: [],
      viewPerson: jest.fn(),
    }
    const tree = render(<RecentResults {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it(`renders a filled RecentResults component`, async () => {
    const props = {
      clearRecentResults: jest.fn(),
      recents: [
        {
          department: `Department of Agriculture`,
          email: `potato@agri.uk`,
          name: `Mr Potato`,
        },
        {
          department: `Faculty of Medical Sciences`,
          email: `chris@house.md`,
          name: `Chris P Bacon`,
        },
        {
          department: `Philosophy Department`,
          email: `jerry.b@panopticon.ac.uk`,
          name: `Jeremy Bentham`,
        },
      ],
      viewPerson: jest.fn(),
    }
    const wrapper = render(<RecentResults {...props} />)
    expect(wrapper).toMatchSnapshot()

    const { getByText } = wrapper
    const result = getByText(props.recents[0].name)
    fireEvent.press(result)

    expect(props.viewPerson).toHaveBeenCalledWith(props.recents[0].email)
  })
})
