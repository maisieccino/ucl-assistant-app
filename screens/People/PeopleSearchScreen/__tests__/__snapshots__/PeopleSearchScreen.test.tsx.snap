// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentResults renders a filled RecentResults component 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu-bold",
                "fontSize": 28,
                "marginBottom": 2,
                "marginTop": 8,
              },
              undefined,
            ]
          }
        >
          Recently Searched
        </Text>
        <TouchableOpacity
          activeOpacity={0.9}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "marginBottom": 10,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1845,
                  "shadowRadius": 1.62,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#eaeaea",
                  "borderRadius": 80,
                  "marginRight": 10,
                  "padding": 10,
                }
              }
            >
              <Text />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu-bold",
                      "fontSize": 16,
                      "marginBottom": 0,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                Mr Potato
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#6E707F",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "fontStyle": "italic",
                      "marginBottom": 1,
                      "marginTop": 0,
                    },
                    undefined,
                  ]
                }
              >
                Department of Agriculture
              </Text>
            </View>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.9}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "marginBottom": 10,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1845,
                  "shadowRadius": 1.62,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#eaeaea",
                  "borderRadius": 80,
                  "marginRight": 10,
                  "padding": 10,
                }
              }
            >
              <Text />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu-bold",
                      "fontSize": 16,
                      "marginBottom": 0,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                Chris P Bacon
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#6E707F",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "fontStyle": "italic",
                      "marginBottom": 1,
                      "marginTop": 0,
                    },
                    undefined,
                  ]
                }
              >
                Faculty of Medical Sciences
              </Text>
            </View>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.9}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "marginBottom": 10,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "padding": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1845,
                  "shadowRadius": 1.62,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#eaeaea",
                  "borderRadius": 80,
                  "marginRight": 10,
                  "padding": 10,
                }
              }
            >
              <Text />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu-bold",
                      "fontSize": 16,
                      "marginBottom": 0,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                Jeremy Bentham
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#6E707F",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "fontStyle": "italic",
                      "marginBottom": 1,
                      "marginTop": 0,
                    },
                    undefined,
                  ]
                }
              >
                Philosophy Department
              </Text>
            </View>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279998859,
                  4283151357,
                ],
                "endPoint": Array [
                  1,
                  0,
                ],
                "locations": undefined,
                "startPoint": Array [
                  0,
                  1,
                ],
              }
            }
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "elevation": 2,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                  "marginTop": 5,
                  "minHeight": 55,
                  "minWidth": 150,
                  "padding": 8,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "shadowColor": "#1B998B",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.75,
                  "shadowRadius": 3,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fefefe",
                    "fontFamily": "apercu",
                    "fontSize": 20,
                    "marginBottom": 1,
                    "marginTop": 1,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Clear
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </TouchableOpacity>
      </View>
    </View>
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#4D5061",
              "fontFamily": "apercu-bold",
              "fontSize": 28,
              "marginBottom": 2,
              "marginTop": 8,
            },
            undefined,
          ]
        }
      >
        Recently Searched
      </Text>
      <TouchableOpacity
        activeOpacity={0.9}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "marginBottom": 10,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1845,
                "shadowRadius": 1.62,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eaeaea",
                "borderRadius": 80,
                "marginRight": 10,
                "padding": 10,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu-bold",
                    "fontSize": 16,
                    "marginBottom": 0,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Mr Potato
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#6E707F",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "fontStyle": "italic",
                    "marginBottom": 1,
                    "marginTop": 0,
                  },
                  undefined,
                ]
              }
            >
              Department of Agriculture
            </Text>
          </View>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.9}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "marginBottom": 10,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1845,
                "shadowRadius": 1.62,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eaeaea",
                "borderRadius": 80,
                "marginRight": 10,
                "padding": 10,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu-bold",
                    "fontSize": 16,
                    "marginBottom": 0,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Chris P Bacon
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#6E707F",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "fontStyle": "italic",
                    "marginBottom": 1,
                    "marginTop": 0,
                  },
                  undefined,
                ]
              }
            >
              Faculty of Medical Sciences
            </Text>
          </View>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.9}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "marginBottom": 10,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1845,
                "shadowRadius": 1.62,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eaeaea",
                "borderRadius": 80,
                "marginRight": 10,
                "padding": 10,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu-bold",
                    "fontSize": 16,
                    "marginBottom": 0,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Jeremy Bentham
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#6E707F",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "fontStyle": "italic",
                    "marginBottom": 1,
                    "marginTop": 0,
                  },
                  undefined,
                ]
              }
            >
              Philosophy Department
            </Text>
          </View>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279998859,
                4283151357,
              ],
              "endPoint": Array [
                1,
                0,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                1,
              ],
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "marginTop": 5,
                "minHeight": 55,
                "minWidth": 150,
                "padding": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "shadowColor": "#1B998B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 3,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontFamily": "apercu",
                  "fontSize": 20,
                  "marginBottom": 1,
                  "marginTop": 1,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Clear
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </TouchableOpacity>
    </View>
  </View>,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RecentResults renders an empty RecentResults component 1`] = `
Object {
  "asJSON": [Function],
  "baseElement": <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>,
  "container": <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />,
  "debug": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByHintText": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByDisplayValue": [Function],
  "findByHintText": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByHintText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByDisplayValue": [Function],
  "getByHintText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByHintText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByDisplayValue": [Function],
  "queryByHintText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
